amazon-s3/5.6/modules/ROOT/pages/amazon-s3-connector-config-topics.adoc

```diff
@@ -9,6 +9,151 @@ In the `Create Object` operation, set the `Content Length` to a value greater th
 
 To encrypt objects that you want to store in S3 buckets using customer managed master keys, specify a Customer Master Key ID in the `KMS Master Key` field in the `Create Object` configuration.
 
+== Prerequisites for Source Operations
+
+There are two scenarios for the `On New Object` and `On Deleted Object` source operations:
+
+. The <<queue-is-created,queue is created>> and specified
+. The <<queue-created-by-source,queue will be created>> by the source
+
+[[queue-is-created]]
+== Queue is Already Created
+
+To correctly use a source operation that already exists, you must enable these actions for the Amazon SQS queue:
+
+[source,json,linenums]
+----
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": [
+                "sqs:DeleteMessage",
+                "sqs:GetQueueUrl",
+                "sqs:ReceiveMessage",
+                "sqs:GetQueueAttributes"
+            ],
+            "Resource": "arn:aws:sqs:REGION:ACCOUNT_ID:QUEUE_NAME"
+        }
+    ]
+}
+----
+
+Additionally, to enable the queue to receive messages when a new object is created in an S3 bucket, you must create permission for the action in this format:
+
+[source,json,linenums]
+----
+{
+  "Version": "2012-10-17",
+  "Id": "arn:aws:sqs:REGION:ACCOUNT_ID:QUEUE_NAME/SQSDefaultPolicy",
+  "Statement": [
+    {
+      "Sid": "Sid1593770403887",
+      "Effect": "Allow",
+      "Principal": "*",
+      "Action": "SQS:SendMessage",
+      "Resource": "arn:aws:sqs:REGION:ACCOUNT_ID:QUEUE_NAME",
+      "Condition": {
+        "ArnLike": {
+          "aws:SourceArn": "arn:aws:s3:::BUCKET_NAME"
+        }
+      }
+    }
+  ]
+}
+----
+
+[[queue-created-by-source]]
+== Queue Created by Source Operation
+
+If the queue is created by the source operation, the queue is created with the prefix `MULE-S3-TRIGGER`. 
+
+In this scenario, you must have permission to create a queue and to also set the queue attributes:
+
+[source,json,linenums]
+----
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": [
+                "sqs:DeleteMessage",
+                "sqs:ReceiveMessage",
+                "sqs:GetQueueAttributes",
+                "sqs:CreateQueue",
+                "sqs:SetQueueAttributes"
+            ],
+            "Resource": "arn:aws:sqs:REGION:ACCOUNT_ID:MULE-S3-TRIGGER-*"
+        }
+    ]
+}
+----
+
+When the queue is created by the source operation, an S3 policy is added to the queue when the S3 object is created in the bucket, in the following format:
```

Comment:
S3 policy -> SQS policy

Reply:
This tech review comment confused me: are we documenting S3 or SQS policies, or both, and do we need to clarify that earlier in this topic set?

Is the policy added automatically?

Is this what we mean:
```suggestion
Queues created by using a source operation automatically receive an SQS policy when the SQS object is created in the bucket, in the following format:
```

Reply:
But...what about the S3 policy referred to in the following section?
I am confused: we show one policy format here, and another below, and it is unclear who creates the policy or creates objects in the bucket. But if this is just my lack of product knowledge, well, you know what to do....  ;-)

Reply:
We are documenting S3 source operations but behind the scene these source operations are using SQS to notify when a new object is created on S3 bucket. S3 policy user needs to add to his created bucket. and we have two scenarios.. 1. queue is already created on aws console therefore user does not need a permission (action) to create queue. 2. If the queue is not created on aws therefore source operation behind the scenes will create a queue and that's why user need a permission (action) to create queue. If the user does not have a permission to create queue in the console he will see access denied.